aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -143,13 +143,37 @@ def setup(self):
         self.ctx.is_converged = False
         self.ctx.iteration = 0
 
+        # Determine the correct `RelaxType` and add it to the context
+        if 'relaxation_scheme' in self.inputs:
+            if self.inputs.relaxation_scheme.value == 'relax':
+                self.ctx.relax_type = RelaxType.ATOMS
+            elif self.inputs.relaxation_scheme.value == 'vc-relax':
+                self.ctx.relax_type = RelaxType.ATOMS_CELL
+            else:
+                raise ValueError('unsupported value for the `relaxation_scheme` input.')
+        else:
+            self.ctx.relax_type = RelaxType(self.inputs.relax_type)
+
+        # Set the meta_convergence and add it to the context
+        self.ctx.meta_convergence = self.inputs.meta_convergence.value
+        if self.inputs.meta_convergence and self.ctx.relax_type in [RelaxType.NONE, RelaxType.ATOMS, RelaxType.SHAPE]:
+            self.report(
+                f'Since there is no change in volume for `{self.ctx.relax_type}`, meta convergence is turned off.'
+            )
+            self.ctx.meta_convergence = False
+
+        # Add the final scf inputs to the context if a final scf should be run
         if self.inputs.final_scf and 'base_final_scf' in self.inputs:
             raise ValueError('cannot specify `final_scf=True` and `base_final_scf` at the same time.')
         elif self.inputs.final_scf:
             self.ctx.final_scf_inputs = AttributeDict(self.exposed_inputs(PwBaseWorkChain, namespace='base'))
         elif 'base_final_scf' in self.inputs:
             self.ctx.final_scf_inputs = AttributeDict(self.exposed_inputs(PwBaseWorkChain, namespace='base_final_scf'))
 
+        if self.ctx.relax_type is RelaxType.NONE:
```

Comment:
This should also include `RelaxType.ATOMS` right? As long as the cell doesn't change, we good.